heron/spi/src/java/com/twitter/heron/spi/utils/SchedulerUtils.java

```diff
@@ -438,4 +438,20 @@ public static boolean curlAndExtractPackage(
 
     return true;
   }
+
+  /**
+   * Get a resource that requires the maximum amount of ram, cpu and disk
+   */
+  public static PackingPlan.Resource getMaxRequiredResource(PackingPlan packingPlan) {
+    PackingPlan.Resource maxResource = new PackingPlan.Resource(0, 0, 0);
+
+    for (PackingPlan.ContainerPlan entry : packingPlan.containers.values()) {
+      PackingPlan.Resource resource = entry.resource;
+      maxResource.ram = Math.max(maxResource.ram, resource.ram);
+      maxResource.cpu = Math.max(maxResource.cpu, resource.cpu);
+      maxResource.disk = Math.max(maxResource.disk, resource.disk);
+    }
+
+    return maxResource;
```

Comment:
Scheduler can update the PackingPlan based on its need. But here, the util method only helps to find the maximum cpu/ram/disk resource requirement.